src/test/java/seedu/address/ui/HomeCardTest.java

```diff
@@ -0,0 +1,149 @@
+        verifyThat(HOME_CARD_SECOND_FX_CODE, (Label label) -> {
+            return label.getText().equals(expectedTime) || label.getText().equals(alternateTime);
+        });
+    }
+
+    @Test
+    public void execute_moneyChartInitialization_success() {
+        // This test ensures the chart is populated with correct data values after initialization
+        verifyThat(HOME_CARD_CHART_FX_CODE, (BarChart<String, Number> chart) -> {
+            return chart.getData().size() == SERIES_SIZE; // Confirm chart has one data series
+        });
+    }
+
+    @Test
+    public void execute_availableTodayListPopulation_success() {
+        // Assumes you want to check if the list correctly displays only those available today
+        Day today = Day.valueOf(LocalDate.now().getDayOfWeek().name());
+        verifyThat(HOME_CARD_AVAILABLE_LIST_FX_CODE, (ListView<Person> listView) -> {
+            return listView.getItems().stream().allMatch(person -> person.getDaysAvailable().contains(today));
+        });
+    }
+
+    @Test
+    public void execute_homeCardInitialization_success() {
+        // Verify that contact count label is correctly set
+        verifyThat(HOME_CARD_CONTACT_FX_CODE, hasText(Integer.toString(model.getFilteredPersonList().size())));
+    }
+
+    @Test
+    public void execute_totalDebtComputation_success() {
+        double expectedTotalDebt = model.getFilteredPersonList().stream()
+                .filter(person -> person.getMoneyOwed().userOwesMoney())
+                .mapToDouble(person -> person.getMoneyOwed().getAbsoluteAmount())
+                .sum();
+
+        assertEquals(expectedTotalDebt, homeCard.getTotalDebt());
+    }
+
+    @Test
+    public void execute_totalCreditComputation_success() {
+        double expectedTotalCredit = model.getFilteredPersonList().stream()
+                .filter(person -> !person.getMoneyOwed().userOwesMoney())
+                .mapToDouble(person -> person.getMoneyOwed().getAbsoluteAmount())
+                .sum();
+
+        assertEquals(expectedTotalCredit, homeCard.getTotalCredit());
+    }
+
+    @Test
+    public void execute_chartConfiguration_success() {
+        verifyThat(HOME_CARD_CHART_FX_CODE, (BarChart<String, Number> chart) -> {
+            return chart.getCategoryGap() == CHART_CATEGORY_GAP
+                    && chart.getBarGap() == CHART_BAR_GAP
+                    && chart.isLegendVisible() == CHART_LEGEND_VISIBLE;
+        });
+    }
+
+    @Test
+    public void execute_moneyChartValues_success() {
+        verifyThat("#chart", (BarChart<String, Number> chart) -> {
+            XYChart.Series<String, Number> series = chart.getData().get(0);
+            return series.getData().get(CHART_DEBT_BAR_INDEX).getYValue().equals(homeCard.getTotalDebt())
+                    && series.getData().get(CHART_CREDIT_BAR_INDEX).getYValue().equals(homeCard.getTotalCredit());
+        });
+    }
+}
```

Comment:
Is there suppose to be an empty line at the bottom for checkstyle?

Reply:
Yes, checkstyle was run for all modified files, thanks for clarifying Alvin!